pkg/ruler/compat.go

```diff
@@ -14,6 +14,7 @@ import (
 	"github.com/weaveworks/common/user"
 
 	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/cortexproject/cortex/pkg/util/validation"
```

Comment:
To avoid introducing unnecessary dependencies, and to simplify writing unit tests, we typically add new interface with the method used by ruler. (Examples: `BlocksStoreLimits`, `MetadataValidationConfig` and few others... all of which `validation.Overrides` implement).

Here we can add `RulesLimits` with `EvaluationDelay(userID)` method.